File: Domain/DataObjects/Collection.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace Domain.DataObjects
{
    public class Collection
    {
        [Key]
        public Guid CollectionId { get; set; }

        // note: this isn't a nav property per se, just a reference to the creator
        public Guid CreatorLanguageProfileId { get; set; }
        public string CreatorUserName { get; set; }
        //Columns
        public bool IsPrivate { get; set; }
        public DateTime CreatedAt { get; set; }
        public string Language { get; set; }
        public string CollectionName { get; set; }
        public string Description { get; set; }
        public ICollection<CollectionContent> CollectionMembers { get; set; }
        public ICollection<SavedCollection> SavedCollections { get; set; }
    }
}

File: codex-client/src/app/models/collection.ts
import { ContentMetadata } from "./content";

export interface Collection {
    collectionId: string,
    creatorLanguageProfileId: string,
    creatorUsername: string,
    isPrivate: boolean,
    createdAt: string,
    language: string,
    collectionName: string,
    description: string,
    contents: ContentMetadata[]
}

export interface CreateCollectionQuery {
    creatorUsername: string,
    isPrivate: boolean,
    language: string,
    collectionName: string,
    description: string,
    firstContentUrl: string
}

export interface CollectionsForLanguageQuery {
    language: string,
    enforceVisibility: boolean
}

export const getCollectionsArray = (map: Map<string, Collection>): Collection[] => {
    let out: Collection[] = [];
    map.forEach((value: Collection, key: string) => {
        out.push(value);
    })
    return out;
}